python/dgl/view.py

```diff
@@ -345,31 +384,72 @@ def __call__(self, *args, **kwargs):
         return self._graph.all_edges(*args, **kwargs)
 
 class HeteroEdgeDataView(MutableMapping):
-    """The data view class when G.ndata[etype] is called."""
-    __slots__ = ['_graph', '_etype', '_etid', '_edges']
+    """The data view class when G.edata[etype] is called."""
+    __slots__ = ['_graph', '_etype', '_multi_etype', '_etid', '_edges']
 
     def __init__(self, graph, etype, edges):
         self._graph = graph
         self._etype = etype
-        self._etid = self._graph.get_etype_id(etype)
+        self._multi_etype = isinstance(etype, list)
+        self._etid = [self._graph.get_etype_id(t) for t in etype] \
+                     if self._multi_etype \
+                     else self._graph.get_etype_id(etype)
         self._edges = edges
 
     def __getitem__(self, key):
-        return self._graph._get_e_repr(self._etid, self._edges)[key]
+        if self._multi_etype:
+            ret = {}
+            for (i, etype) in enumerate(self._etype):
+                value = self._graph._get_e_repr(self._etid[i], self._edges).get(key, None)
+                if value is not None:
+                    ret[etype] = value
+            return ret
+        else:
+            return self._graph._get_e_repr(self._etid, self._edges)[key]
 
     def __setitem__(self, key, val):
-        self._graph._set_e_repr(self._etid, self._edges, {key : val})
+        if self._multi_etype:
+            assert isinstance(val, dict), \
+                'Current HeteroEdgeDataView has multiple edge types, ' \
+                'please passing the edge type and the corresponding data through a dict.'
+
+            for (etype, data) in val.items():
+                etid = self._graph.get_etype_id(etype)
+                self._graph._set_e_repr(etid, self._edges, {key : data})
+        else:
+            self._graph._set_e_repr(self._etid, self._edges, {key : val})
```

Comment:
I think we also need to check if val is a dictionary here and raise an error if so.